vividus-plugin-db/src/main/java/org/vividus/bdd/steps/db/DatabaseSteps.java

```diff
@@ -208,6 +209,56 @@ public void compareData(String sourceSqlQuery, String sourceDbKey, String target
         verifyComparisonResult(queriesStatistic, result);
     }
 
+    /**
+     * Step waits until data received from SQL request is equal to data from examples table for some duration.
+     * Actions performed in the step:
+     * <ul>
+     *     <li>executes provided SQL query against database</li>
+     *     <li>compares data received from SQL request against examples table</li>
+     *     <li>sleeps during calculated part of specified duration</li>
+     *     <li>repeats previous actions if data was not equal and seconds timeout not expired</li>
+     * </ul>
+     * @param duration Time duration to wait
+     * @param retryTimes How many times request will be retried; duration/retryTimes=timeout between requests
+     * @param sqlQuery SQL query to execute
+     * @param dbKey Key identifying the database connection
+     * @param keys Comma-separated list of column's names to map resulting tables rows
+     * @param table Rows to compare data against
+     */
+    @When("I wait for '$duration' duration retrying $retryTimes times while data from `$sqlQuery`"
+            + " executed against `$dbKey` matching rows using `$keys` is equal to data from:$table")
```

Comment:
do you really need
```
matching rows using `$keys`
```
in this step?

Reply:
I thought it's a common matching parameter for comparison steps. Could I remove it?

Reply:
not really common, if you don't rely on this parameter, you can remove it